flake.nix

```diff
@@ -81,13 +86,23 @@
           ln -s ${inputs.tree-sitter-ruby} vendor/tree-sitter-ruby
           ln -s ${inputs.tree-sitter-rust} vendor/tree-sitter-rust
           ln -s ${inputs.tree-sitter-typescript} vendor/tree-sitter-typescript
+          ln -s ${inputs.tree-sitter-nix} vendor/tree-sitter-nix
         '';
       in rec {
         # `nix build`
         packages.tree-grepper = naersk-lib.buildPackage {
-          root = ./.;
+          root = inputs.nix-filter.lib.filter {
+            root = ./.;
+              include = [
+                (inputs.nix-filter.lib.inDirectory ./src)
+                ./build.rs
+                ./Cargo.lock
+                ./Cargo.toml
+              ];
+          };
```

Comment:
I'm sorry if I wasn't clear enough here: I don't want to make any changes in this PR aside from adding the new grammar. Could you please clean out all of this?

Reply:
Yeh, I could.